pxr/usd/usdMtlx/materialXInfoAPI.h

```diff
@@ -0,0 +1,209 @@
+    ///
+    /// \code
+    /// UsdMtlxMaterialXInfoAPI(stage->GetPrimAtPath(path));
+    /// \endcode
+    ///
+    USDMTLX_API
+    static UsdMtlxMaterialXInfoAPI
+    Get(const UsdStagePtr &stage, const SdfPath &path);
+
+
+    /// Returns true if this <b>single-apply</b> API schema can be applied to 
+    /// the given \p prim. If this schema can not be a applied to the prim, 
+    /// this returns false and, if provided, populates \p whyNot with the 
+    /// reason it can not be applied.
+    /// 
+    /// Note that if CanApply returns false, that does not necessarily imply
+    /// that calling Apply will fail. Callers are expected to call CanApply
+    /// before calling Apply if they want to ensure that it is valid to 
+    /// apply a schema.
+    /// 
+    /// \sa UsdPrim::GetAppliedSchemas()
+    /// \sa UsdPrim::HasAPI()
+    /// \sa UsdPrim::CanApplyAPI()
+    /// \sa UsdPrim::ApplyAPI()
+    /// \sa UsdPrim::RemoveAPI()
+    ///
+    USDMTLX_API
+    static bool 
+    CanApply(const UsdPrim &prim, std::string *whyNot=nullptr);
+
+    /// Applies this <b>single-apply</b> API schema to the given \p prim.
+    /// This information is stored by adding "MaterialXInfoAPI" to the 
+    /// token-valued, listOp metadata \em apiSchemas on the prim.
+    /// 
+    /// \return A valid UsdMtlxMaterialXInfoAPI object is returned upon success. 
+    /// An invalid (or empty) UsdMtlxMaterialXInfoAPI object is returned upon 
+    /// failure. See \ref UsdPrim::ApplyAPI() for conditions 
+    /// resulting in failure. 
+    /// 
+    /// \sa UsdPrim::GetAppliedSchemas()
+    /// \sa UsdPrim::HasAPI()
+    /// \sa UsdPrim::CanApplyAPI()
+    /// \sa UsdPrim::ApplyAPI()
+    /// \sa UsdPrim::RemoveAPI()
+    ///
+    USDMTLX_API
+    static UsdMtlxMaterialXInfoAPI 
+    Apply(const UsdPrim &prim);
+
+protected:
+    /// Returns the kind of schema this class belongs to.
+    ///
+    /// \sa UsdSchemaKind
+    USDMTLX_API
+    UsdSchemaKind _GetSchemaKind() const override;
+
+private:
+    // needs to invoke _GetStaticTfType.
+    friend class UsdSchemaRegistry;
+    USDMTLX_API
+    static const TfType &_GetStaticTfType();
+
+    static bool _IsTypedSchema();
+
+    // override SchemaBase virtuals.
+    USDMTLX_API
+    const TfType &_GetTfType() const override;
+
+public:
+    // --------------------------------------------------------------------- //
+    // MTLXINFOVERSION 
+    // --------------------------------------------------------------------- //
+    /// MaterialX library version that the data has been authored against.
+    /// Defaults to 1.38 to allow correct verisoning of old files.
+    ///
+    /// | ||
+    /// | -- | -- |
+    /// | Declaration | `string mtlxinfo:version = "1.38"` |
+    /// | C++ Type | std::string |
+    /// | \ref Usd_Datatypes "Usd Type" | SdfValueTypeNames->String |
+    USDMTLX_API
+    UsdAttribute GetMtlxinfoVersionAttr() const;
```

Comment:
Style question: should "info" and "x" be capitalized in the function @ld-kerley ? (i.e. `GetMtlXInfoVersionAttr()`)

Reply:
Following other MaterialX use in USD I think it should be `Mtlx`, so lower case x.  But I think you're right about the `Info` part - thanks - I'll make sure that gets changed next time I push something here.

Reply:
Cool that's great. Thank you for getting this PR up and going!